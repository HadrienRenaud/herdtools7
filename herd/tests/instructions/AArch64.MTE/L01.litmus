AArch64 S+dmb.sytt+addrpt
Orig=DMB.SYdWWTT RfeTP DpAddrdWPT CoeTT
Variant=mte,async
(* Test is allowed in asynchronous mode, by lack of
   iico_ctrl relation from tag test to read in P1 LDR *)
{
0:X0=x:blue; 0:X1=x:red; 0:X2=y:red; 0:X3=y:green;
1:X0=y:red; 1:X3=x:red; 1:X4=x:green;
}
 P0          | P1                ;
 STG X0,[X1] | LDR W1,[X0]       ;
 DMB SY      | EOR W2,W1,W1      ;
 STG X2,[X3] | ADD X5,X4,W2,SXTW ;
             | STG X3,[X5]       ;
exists (1:X1=0 /\ [tag(x)]=:blue /\ ~fault(P1,y))
