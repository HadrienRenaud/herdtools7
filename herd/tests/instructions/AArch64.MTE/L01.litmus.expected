Test S+dmb.sytt+addrpt Allowed
States 4
1:X1=0; [tag(x)]=:blue;  ~Fault(P1,y);
1:X1=0; [tag(x)]=:blue; Fault(P1,y:red,TagCheck);
1:X1=0; [tag(x)]=:red;  ~Fault(P1,y);
1:X1=0; [tag(x)]=:red; Fault(P1,y:red,TagCheck);
Ok
Witnesses
Positive: 1 Negative: 3
Condition exists (1:X1=0 /\ [tag(x)]=:blue /\ not (fault(P1,y)))
Observation S+dmb.sytt+addrpt Sometimes 1 3
Hash=e590f3a6e742d25cc38d360f8014889e

